doc/source/whatsnew/v0.22.0.txt

```diff
@@ -31,6 +31,36 @@ Other Enhancements
 Backwards incompatible API changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+Better pretty-printing of DataFrames in a terminal
```

Comment:
I think we need a separate entry for the 'number of rows' change, as that is the change with much broader impact on users (I think), and is currently not reflected from this title (eg it is not only for terminal). 
I would explain it as two separate things.